### PR TITLE
Build examples from CMake rather than executable

### PR DESCRIPTION
**Pull request by mjcarroll**
# 🦟 Bug fix

## Summary

Part of https://github.com/gazebosim/gz-cmake/pull/301

Test by building gz-common as normal and then running the "EXAMPLES_Configure_TEST" and "EXAMPLES_Build_TEST" via ctest.

## Checklist
- [ ] Signed all commits for DCO
- [ ] Added tests
- [ ] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [ ] Consider updating Python bindings (if the library has them)
- [ ] `codecheck` passed (See [contributing](https://gazebosim.org/docs/all/contributing#contributing-code))
- [ ] All tests passed (See [test coverage](https://gazebosim.org/docs/all/contributing#test-coverage))
- [ ] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Agazebosim+archived%3Afalse+) to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge** and edit the commit message to match the pull request summary while retaining `Signed-off-by` messages.

### CODE REVIEW

File: examples/CMakeLists.txt [hunk 1]
```diff
@@ -1,4 +1,6 @@
-cmake_minimum_required(VERSION 2.8 FATAL_ERROR)
+cmake_minimum_required(VERSION 3.5 FATAL_ERROR)
+
+project(gz-math-examples)
```
Comment (iche033) [inline examples/CMakeLists.txt:3]: `gz-common-examples`?

**Review by ahcorde — CHANGES_REQUESTED** [review]
`gz_build_examples` is not available. @mjcarroll do you need a `gz-cmake` release?

**Comment by mjcarroll** [comment]
@osrf-jenkins retest this please

**Comment by mjcarroll** [comment]
@osrf-jenkins retest this please

**Review by ahcorde — CHANGES_REQUESTED** [review]
macOS is faling

**Comment by mjcarroll** [comment]
> macOS is faling

Addressed in https://github.com/gazebosim/gz-common/pull/524

**Review by ahcorde — APPROVED** [review]

**Comment by iche033** [comment]
macOS builds are failing to find gz-common5 when building examples. Is that expected?

**Comment by mjcarroll** [comment]
> macOS builds are failing to find gz-common5 when building examples. Is that expected?

We either need to run `make install` before `make test` or skip the examples.  I would vote for the former as it exercises the examples.

**Pull request merged**